lib/readline.js

```diff
@@ -238,6 +243,24 @@ Interface.prototype.setPrompt = function(prompt) {
   this._prompt = prompt;
 };
 
+Interface.prototype.read = function() {
+  if (this._buffer.length === 0)
+    return null;
+
+  return this._buffer.shift();
+};
+
+
+Interface.prototype[Symbol.asyncIterator] = function() {
+  emitExperimentalWarning('readline Interface[Symbol.asyncIterator]');
+
+  this._enableBuffer = true;
```

Comment:
I don't see this ever getting restored after the completion of iteration, and I feel like it should be. Correct me if I'm wrong :)